### DISCUSSION TITLE
Why would I get this build error?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#39625] [Help] [answered]
```
info  - Creating an optimized production build
Failed to compile.

./pages/_document.js
```

Obviously isn't compiling the _document.js..... but doesn't tell me why? 

I am trying to convert this site into a PWA.... seems pretty easy, but maybe I am missing information????    What else would I need to include? 

The pages I have added are these:
https://github.com/Elindo586/nextprojectb/blob/main/pages/_document.js

A few icons and manifest.json
https://github.com/Elindo586/nextprojectb/blob/main/public/manifest.json

Next config... I think this might be wrong:
https://github.com/Elindo586/nextprojectb/blob/main/next.config.js

Overall seems to be pretty simple...  but why would it be giving me errors?

**Comment by gaston-flores** [comment, marked as answer]
You have a typo in your imports:
```jsx
import Document, { Html, Head, Main, NextScript } from "next/document,";
```

Remove the extra comma at the end of `next/document`. Also, you are not exporting your custom `Document` from `pages/_document.js`. Add an export declaration at the end of the file:

```jsx
import Document, { Html, Head, Main, NextScript } from "next/document";

class MyDocument extends Document {
  render() {
    return (
      <Html>
        <Head>
          <link rel="manifest" href="/manifest.json " />
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}

export default MyDocument;
```

Finally, keep in mind you'll need to add your `nextConfig` to `withPWA` or it will be ignored.

**Comment by Elindo586** [comment]
Ah thanks... that worked I think...

Would you know why I would be getting this from the localhost 3000?

```
Unhandled Runtime Error
TypeError: Failed to register a ServiceWorker for scope [('http://localhost:3000/')](http://localhost:3000/('http://localhost:3000/')) with script [('http://localhost:3000/sw.js'):](http://localhost:3000/('http://localhost:3000/sw.js'):) A bad HTTP response code (404) was received when fetching the script.
```

Also... the nextconfig looks like this:
```
/** @type {import('next').NextConfig} */

const withPWA = require("next-pwa");

const nextConfig = {
  reactStrictMode: true,
  async rewrites() {
    return {
      afterFiles: [{ source: "/:path*", destination: "/_404/:path*" }],
    };
  },
};

module.exports = withPWA({
  pwa: {
    dest: "public",
    register: true,
    skipWaiting: true,
    disable: process.env.NODE_ENV === "development",
  },
});


```

**Reply by Elindo586** [reply]
Hmmmm... the redirects stopped working...

**Reply by gaston-flores** [reply]
Yep, that' because `nextConfig` is not really doing anything right now. Add the values to the withPWA object or spread `nextConfig` inside it.

```js
/** @type {import('next').NextConfig} */

const withPWA = require("next-pwa");

const nextConfig = {
  reactStrictMode: true,
  async rewrites() {
    return {
      afterFiles: [{ source: "/:path*", destination: "/_404/:path*" }],
    };
  },
};

module.exports = withPWA({
  ...nextConfig,
  pwa: {
    dest: "public",
    register: true,
    skipWaiting: true,
    disable: process.env.NODE_ENV === "development",
  },
});
```

**Comment by Elindo586** [comment]
Oh... I am also getting a bunch of warning errors... 

is the title and viewport not supposed to be in the _document.js? 

I was following this example here.... I guess this example is outdated? 
https://github.com/vercel/next.js/blob/canary/examples/progressive-web-app/pages/_app.tsx

**Comment by Elindo586** [comment]
I deleted the title and viewport from _documents.js


Right now I am checking this error:

`warn  - GenerateSW has been called multiple times, perhaps due to running webpack in --watch mode. The precache manifest generated after the first call may be inaccurate! Please see https://github.com/GoogleChrome/workbox/issues/1790 for more information.`

**Reply by gaston-flores** [reply]
Seems to be a problem related to https://github.com/shadowwalker/next-pwa/issues/371, it should only be an issue in development.
